standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -4026,6 +4031,41 @@ public List<String> listPartitionNamesPs(String catName, String dbName, String t
     return partitionNames;
   }
 
+  private List<MPartition> listMPartitionsByNames(String catName, String dbName, String tableName,
+      List<String> partNames) throws Exception {
+    LOG.debug("Executing listMPartitionsByNames");
+
+    Preconditions.checkState(this.currentTransaction.isActive());
```

Comment:
Can you refactor common code from listMPartitionsByNames and listMPartitions methods

Reply:
Not using the same API now, changed the code.